### PR TITLE
UI: Friendly driver name reporting.

### PR DESCRIPTION
**Pull request by MutantAura**
Adds a method to pull a more friendly driver name from the `VkDriverId`. MacOS will prefer vendor name for the time being.
Needs testing on Linux systems mainly as the drivers there are significantly more varied. Feedback on the friendly strings is also welcome.

Potentially supersedes https://github.com/Ryujinx/Ryujinx/pull/6208.

### CODE REVIEW

**Comment by dnmodder** [comment]
AMDVLK:
Captura de pantalla_20240319_180542

RADV:
Captura de pantalla_20240319_180806

Looks good.

**Review by IsaacMarovitz — APPROVED** [review]

File: src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs [hunk 1]
```diff
@@ -751,8 +749,9 @@ private unsafe void PrintGpuInformation()
                 Vendor == Vendor.Broadcom ||
                 Vendor == Vendor.ImgTec;
 
-            GpuVendor = vendorName;
-            GpuDriver = hasDriverProperties ? Marshal.PtrToStringAnsi((IntPtr)driverProperties.DriverName) : vendorName; // Fall back to vendor name if driver name isn't available.
+            GpuVendor = VendorUtils.GetNameFromId(properties.VendorID);
+            GpuDriver = hasDriverProperties && !OperatingSystem.IsMacOS() ?
+                VendorUtils.GetFriendlyDriverName(driverProperties.DriverID) : GpuVendor; // Fallback to vendor name if driver is unavailable or on MacOS where vendor is prefered.
```
Comment (gdkchan) [inline src/Ryujinx.Graphics.Vulkan/VulkanRenderer.cs:754]:
```suggestion
                VendorUtils.GetFriendlyDriverName(driverProperties.DriverID) : GpuVendor; // Fallback to vendor name if driver is unavailable or on MacOS where vendor is preferred.
```

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Review by riperiperi — APPROVED** [review]
Nice,

**Comment by gdkchan** [comment]
Needs a rebase.

**Comment by MutantAura** [comment]
Done

**Pull request merged**